internal/otel/otel.go

```diff
@@ -48,3 +59,33 @@ func InitOpenTelemetry(ctx context.Context, logger *logrus.Logger, gwConfig *Con
 
 	return provider
 }
+
+// Span attributes related functions
+func ApidefSpanAttributes(apidef *apidef.APIDefinition) []SpanAttribute {
+	attrs := []SpanAttribute{
+		semconv.TykAPIName(apidef.Name),
+		semconv.TykAPIOrgID(apidef.OrgID),
+		semconv.TykAPIID(apidef.APIID),
+		semconv.TykAPIListenPath(apidef.Proxy.ListenPath),
+	}
+
+	if !apidef.TagsDisabled {
+		tags := apidef.Tags
+		tags = append(tags, apidef.TagHeaders...)
+
+		attrs = append(attrs, semconv.TykAPITags(tags...))
+	}
+
+	return attrs
+}
+
+func APIVersionAttribute(version string) SpanAttribute {
+	if version == "" {
+		version = "Non Versioned"
```

Comment:
what about moving this string ("Non Versioned") into a constant?

Reply:
Done!